Code/GraphMol/MolDraw2D/MolDraw2D.cpp

```diff
@@ -528,13 +530,10 @@ void MolDraw2D::drawReaction(const ChemicalReaction &rxn,
   std::vector<int> *bond_highlights = NULL;
   std::map<int, DrawColour> *bond_highlight_colors = NULL;
   if (highlightByReactant) {
-    // FIX: these really should be defined elsewhere
-    std::vector<DrawColour> highlightColors;
-    highlightColors.push_back(DrawColour(.9, .6, .6));
-    highlightColors.push_back(DrawColour(.6, .6, .9));
-    highlightColors.push_back(DrawColour(.8, .8, .3));
-    highlightColors.push_back(DrawColour(.8, .6, .8));
-
+    std::vector<DrawColour> colors = drawOptions().highlightColourPalette;
```

Comment:
what about making `colors` a `std::vector<DrawColour> *` so that you can skip all the copying?